### PR TITLE
[Docs] gpl documentation

### PR DESCRIPTION
**Pull request by luarss**
Update missing parameters and formatting.

Also removal of this page which is a repetition
image

https://openroad--3702.org.readthedocs.build/en/3702/main/src/gpl/README.html

### CODE REVIEW

File: src/gpl/README.md [hunk 1]
```diff
@@ -1,6 +1,8 @@
-# RePlAce
+# Global Placement
 
-RePlAce: Advancing Solution Quality and Routability Validation in Global Placement
+OpenROAD performs global placement using the RePlAce algorithm, from the
```
Comment (vvbandeira) [inline src/gpl/README.md:3]: See `drt` PR for the correct phrasing.

File: src/gpl/README.md [hunk 2]
```diff
@@ -51,87 +68,111 @@ global_placement
...
+| `-timing_driven_net_reweight_overflow` | Set overflow threshold for timing-driven net reweighting. Allowed values are `tcl list of [0-100, int]`. |
+| `-timing_driven_net_weight_max` | Set the multiplier for the most timing critical nets. Default value is 1.9. |
+| `-timing_driven_nets_percentage` | Set the percentage of nets that are reweighted in timing-driven mode. Default value is 10. Allowed values are `[0-100, float]`. |
+| `-verbose_level` | Set verbose level for RePlAce. Default value is 1. Allowed values are `[0-5, int]`. |
+| `-force_cpu` | Force to use the CPU solver even if the GPU is available. | 
+
+
+### Useful developer functions
+
+If you are a developer, you might find these useful. More details can be found in the [source file](./src/replace.cpp) or the [swig file](./src/replace.i).
```
Comment (vvbandeira) [inline src/gpl/README.md:97]: In `drt` PR these links are full `https...` which is correct? Please use the same format in all docs.

File: src/gpl/README.md [hunk 3]
```diff
@@ -51,87 +68,111 @@ global_placement
...
+Example scripts demonstrating how to run RePlAce on a sample design on `core01` as follows:
 
-## Example scripts
+```shell
+./test/core01.tcl
+```
 
 ## Regression tests
 
+There are a set of regression tests in `/test`. For more information, refer to this [section](../../README.md#regression-tests).
```
Comment (vvbandeira) [inline src/gpl/README.md:117]:
```suggestion
There are a set of regression tests in `./test`. For more information, refer to this [section](../../README.md#regression-tests).
```

File: src/gpl/README.md [hunk 4]
```diff
@@ -18,6 +20,21 @@ Features:
...
 ## Commands
 
+```{note}
+- Parameters in square brackets `[-param param]` are optional.
+- Parameters without square brackets `-param2 param2` are required.
+```
+
+### Global Placement
+
+`-timing_driven` does a virtual `repair_design` to find slacks and
```
Comment (vvbandeira) [inline src/gpl/README.md:30]:
```suggestion
When using the `-timing_driven` flag, `gpl` does a virtual `repair_design` to find slacks and
```

File: src/gpl/README.md [hunk 5]
```diff
@@ -51,87 +68,111 @@ global_placement
...
-- `-initial_place_max_iter`: set maximum iterations in initial place. Default value is 20. Allowed values are `[0-MAX_INT, int]`.
-- `-initial_place_max_fanout`: set net escape condition in initial place when 'fanout >= initial_place_max_fanout'. Default value is 200. Allowed values are `[1-MAX_INT, int]`.
-- `-timing_driven_net_reweight_overflow`: set overflow threshold for timing-driven net reweighting. Allowed values are `tcl list of [0-100, int]`.
-- `-timing_driven_net_weight_max`: Set the multiplier for the most timing critical nets. Default value is 1.9.
-- `-timing_driven_nets_percentage`: Set the percentage of nets that are reweighted in timing-driven mode. Default value is 10. Allowed values are `[0-100, float]`
-- `-verbose_level`: set verbose level for RePlAce. Default value is 1. Allowed values are `[0-5, int]`.
-- `-force_cpu`: Force to use the CPU solver even if the GPU is available.
+#### Options
+
+| Switch Name | Description |
```
Comment (vvbandeira) [inline src/gpl/README.md:73]: This is a long table; let's break it into three tables. One for arguments independent of `timing/routability` driven and one for each `timing/routability` only arguments. Please also order the arguments in `replace.tcl` to reflect this grouping.

File: src/gpl/README.md [hunk 6]
```diff
@@ -51,87 +68,111 @@ global_placement
...
+| ----- | ----- |
+| `-timing_driven` | Enable timing-driven mode. |
+| `-routability_driven` | Enable routability-driven mode. |
+| `-skip_initial_place` | Skip the initial placement (Biconjugate gradient stabilized, or BiCGSTAB solving) before Nesterov placement. Initial placement improves HPWL by ~5% on large designs. Equivalent to `-initial_place_max_iter 0`. | 
+| `-incremental` | Enable the incremental global placement. Users would need to tune other parameters (e.g., `init_density_penalty`) with pre-placed solutions. | 
+| `-bin_grid_count` | Set bin grid's counts. Default value is defined by internal heuristic. Allowed values are  `[64,128,256,512,..., int]`. |
+| `-density` | Set target density. Default value is 0.70. Allowed values are `[0-1, float]`. |
```
Comment (vvbandeira) [inline src/gpl/README.md:80]:
```suggestion
| `-density` | Set target density. The default value is `0.7` (i.e., 70%). Allowed values are floats `[0, 1]`. |
```

File: src/gpl/README.md [hunk 7]
```diff
@@ -51,87 +68,111 @@ global_placement
...
+| `-init_density_penalty` | Set initial density penalty. Default value is 8e-5. Allowed values are `[1e-6 - 1e6, float]`. |
+| `-init_wirelength_coef` | Set initial wirelength coefficient. Default value is 0.25. Allowed values are `[unlimited, float]`. |
+| `-min_phi_coef` | Set `pcof_min` ($\mu_k$ Lower Bound). Default value is 0.95. Allowed values are `[0.95-1.05, float]`. |
+| `-max_phi_coef` | Set `pcof_max` ($\mu_k$ Upper Bound). Default value is 1.05. Allowed values are `[1.00-1.20, float]`. |
```
Comment (vvbandeira) [inline src/gpl/README.md:84]: Developer options?

File: src/gpl/README.md [hunk 8]
```diff
@@ -51,87 +68,111 @@ global_placement
...
+| `-overflow` | Set target overflow for termination condition. Default value is 0.1. Allowed values are `[0-1, float]`. |
```
Comment (vvbandeira) [inline src/gpl/README.md:85]:
```suggestion
| `-overflow` | Set target overflow for termination condition. The default value is `0.1`. Allowed values are floats `[0, 1]`. |
```

File: src/gpl/README.md [hunk 9]
```diff
@@ -51,87 +68,111 @@ global_placement
...
+| `-initial_place_max_iter` | Set maximum iterations in initial place. Default value is 20. Allowed values are `[0-MAX_INT, int]`. |
```
Comment (vvbandeira) [inline src/gpl/README.md:86]:
```suggestion
| `-initial_place_max_iter` | Set maximum iterations in the initial place. The default value is 20. Allowed values are integers `[0, MAX_INT]`. |
```

File: src/gpl/README.md [hunk 10]
```diff
@@ -51,87 +68,111 @@ global_placement
...
-- `-force_cpu`: Force to use the CPU solver even if the GPU is available.
+#### Options
+
+| Switch Name | Description |
+| ----- | ----- |
+| `-timing_driven` | Enable timing-driven mode. |
+| `-routability_driven` | Enable routability-driven mode. |
+| `-skip_initial_place` | Skip the initial placement (Biconjugate gradient stabilized, or BiCGSTAB solving) before Nesterov placement. Initial placement improves HPWL by ~5% on large designs. Equivalent to `-initial_place_max_iter 0`. | 
+| `-incremental` | Enable the incremental global placement. Users would need to tune other parameters (e.g., `init_density_penalty`) with pre-placed solutions. | 
+| `-bin_grid_count` | Set bin grid's counts. Default value is defined by internal heuristic. Allowed values are  `[64,128,256,512,..., int]`. |
```
Comment (vvbandeira) [inline src/gpl/README.md:79]:
```suggestion
| `-bin_grid_count` | Set bin grid's counts. The internal heuristic defines the default value. Allowed values are integers `[64,128,256,512,...]`. |
```

File: src/gpl/README.md (hunk 6 above)
Comment (vvbandeira) [reply src/gpl/README.md:80]: Let's use this notation: "floats/integers/etc `[x, y]`" Remember to use `[ | ]` for inclusive and `( | )` for excluding the first/last number.

File: src/gpl/README.md [hunk 11]
```diff
@@ -51,87 +68,111 @@ global_placement
...
+| `-max_phi_coef` | Set `pcof_max` ($\mu_k$ Upper Bound). Default value is 1.05. Allowed values are `[1.00-1.20, float]`. |
+| `-overflow` | Set target overflow for termination condition. Default value is 0.1. Allowed values are `[0-1, float]`. |
+| `-initial_place_max_iter` | Set maximum iterations in initial place. Default value is 20. Allowed values are `[0-MAX_INT, int]`. |
+| `-initial_place_max_fanout` | Set net escape condition in initial place when $fanout \geq initial\_place\_max\_fanout$. Default value is 200. Allowed values are `[1-MAX_INT, int]`. |
+| `-timing_driven_net_reweight_overflow` | Set overflow threshold for timing-driven net reweighting. Allowed values are `tcl list of [0-100, int]`. |
+| `-timing_driven_net_weight_max` | Set the multiplier for the most timing critical nets. Default value is 1.9. |
+| `-timing_driven_nets_percentage` | Set the percentage of nets that are reweighted in timing-driven mode. Default value is 10. Allowed values are `[0-100, float]`. |
+| `-verbose_level` | Set verbose level for RePlAce. Default value is 1. Allowed values are `[0-5, int]`. |
+| `-force_cpu` | Force to use the CPU solver even if the GPU is available. |
```
Comment (vvbandeira) [inline src/gpl/README.md:92]:
```suggestion
| `-initial_place_max_fanout` | Set net escape condition in initial place when $fanout \geq initial\_place\_max\_fanout$. The default value is 200. Allowed values are integers `[1, MAX_INT]`. |
| `-timing_driven_net_reweight_overflow` | Set overflow threshold for timing-driven net reweighting. Allowed value is a Tcl list of integers where each number is `[0, 100]`. |
| `-timing_driven_net_weight_max` | Set the multiplier for the most timing-critical nets. The default value is `1.9`. |
| `-timing_driven_nets_percentage` | Set the reweighted percentage of nets in timing-driven mode. The default value is 10. Allowed values are floats `[0, 100]`. |
| `-verbose_level` | Set verbose level for `gpl`. The default value is 1. Allowed values are integers `[0, 5]`. |
| `-force_cpu` | Force to use the CPU solver even if the GPU is available. |
```

**Review by vvbandeira — CHANGES_REQUESTED** [review]

File: src/gpl/README.md (hunk 7 above)
Comment (luarss) [reply src/gpl/README.md:84]: I am thinking it can potentially be tuned using autotuner?

Comment (vvbandeira) [reply src/gpl/README.md:86]: Maybe we should consider using `Advanced Options` instead of `Developer Options`. Then, we can include AutoTuner + Dev.

Comment (luarss) [reply src/gpl/README.md:86]: Lets discuss this tomorrow with @dralabeing

File: src/gpl/README.md [hunk 12]
```diff
@@ -1,6 +1,9 @@
-# RePlAce
+# Global Placement
 
-RePlAce: Advancing Solution Quality and Routability Validation in Global Placement
+The global placement module in OpenROAD (`gpl`) is based on the open-source
+RePlAce algorithm, from the
```
Comment (vvbandeira) [inline src/gpl/README.md:4]:
```suggestion
RePlAce tool, from the
```

File: src/gpl/README.md [hunk 13]
```diff
@@ -45,93 +68,133 @@ global_placement
...
-default, modified with `-timing_driven_net_weight_max`). The multiplier
-is scaled from the full value for the worst slack, to 1.0 at the
-timing_driven_nets_percentage point. Use the `set_wire_rc` command to set
-resistance and capacitance of estimated wires used for timing.
+#### General Arguments
+
+| Switch Name | Description |
+| ----- | ----- |
+| `-timing_driven` | Enable timing-driven mode. |
+| `-routability_driven` | Enable routability-driven mode. |
```
Comment (vvbandeira) [inline src/gpl/README.md:78]: Let's add here a "See [link] for timing/routability specific arguments."

**Review by vvbandeira — CHANGES_REQUESTED** [review]

File: docs/toc.yml [hunk 14]
```diff
@@ -54,9 +54,8 @@ entries:
         title: VSRC Description
   - file: main/src/gpl/README
     title: Global Placement
-    entries:
-    - file: main/src/gpl/doc/TclCommands
-      title: Tcl Commands
+  - file: main/src/sta/README
+    title: Static Timing Analyzer (external)
```
Comment (vvbandeira) [inline docs/toc.yml:58]: RTDs is complaining about this and refusing to build. @luarss

**Comment by luarss** [comment]
Addressed https://github.com/The-OpenROAD-Project/OpenROAD/discussions/3962

**Comment by luarss** [comment]
@vvbandeira Pinging for review

File: src/gpl/README.md [hunk 15]
```diff
@@ -94,61 +138,86 @@ cluster_flops
...
-#### Options
+#### General Arguments
+
+| Switch Name | Description |
+| ----- | ----- |
+| `-tray_weight` | Set the weighting factor for tray cost (recommended to be `[20.0, float]`). |
+| `-timing_weight` | Set the weighting factor for timing-critical paths in (recommended to be `[1.0. float]`). |
+| `-max_split_size` | The maximum size of a single pointset after running the pointset decomposition algorithm for runtime improvement (to not run pointset decomposition, set as `-1`). |
+
+# Example Scripts
```
Comment (vvbandeira) [inline src/gpl/README.md:149]:
```suggestion
## Example Scripts
```

**Review by vvbandeira — APPROVED** [review]

**Pull request merged**